File: 2022-1-elixir/test/day_05_test.exs
defmodule Day05Test do
  use ExUnit.Case

  setup_all do
    IO.puts("--- DAY 05 ----")
    {:ok, test_data} = File.read("data/day_05_test.txt")
    {:ok, real_data} = File.read("data/day_05.txt")
    {:ok, test_data: test_data, real_data: real_data}
  end

  describe "part 1" do
    test "test data", context do
      test_result = Day05.part_1(context[:test_data])
      IO.puts("part 1 :: test : #{test_result}")
      assert test_result == "CMZ"
    end

    test "real data", context do
      real_result = Day05.part_1(context[:real_data])
      IO.puts("part 1 :: real : #{real_result}")
      assert real_result
      assert real_result == "CFFHVVHNC"
    end
  end

  describe "part 2" do
    test "test data", context do
      test_result = Day05.part_2(context[:test_data])
      IO.puts("part 2 :: test : #{test_result}")
      assert test_result == "MCD"
    end

    @tag :skip
    test "real data", context do
      real_result = Day05.part_2(context[:real_data])
      IO.puts("part 2 :: real : #{real_result}")
      assert real_result
      assert real_result == "FSZWBPTBG"
    end
  end
end

File: 2024/day_05_test.rb
# frozen_string_literal: true

require './test_helper'
require './day_05'

describe Day05 do
  before do
    @data = StringIO.new(<<~TEXT)
    47|53
    97|13
    97|61
    97|47
    75|29
    61|13
    75|53
    29|13
    97|29
    53|29
    61|53
    97|53
    61|29
    47|13
    75|47
    97|75
    47|61
    75|61
    47|29
    75|13
    53|13

    75,47,61,53,29
    97,61,53,29,13
    75,29,13
    75,97,47,61,53
    61,13,29
    97,13,75,29,47
    TEXT
  end

  it 'returns 143 for part 1' do
    assert_equal 143, Day05.new(io: @data).part_1!
  end

  it 'returns 123 for part 2' do
    assert_equal 123, Day05.new(io: @data).part_2!
  end
end
